policy/release/git_branch/git_branch.rego

```diff
@@ -0,0 +1,27 @@
+package git_branch
+
+import rego.v1
+import data.lib
+
+# METADATA
+# title: Only allow builds from a trusted branch
+# description: Build must originate from a configured branch pattern (e.g., 'refs/heads/main')
+# custom:
+#   short_name: git_protected_branch_required
+#   failure_msg: Build is not from a trusted branch
+#   collections:
+#   - redhat_rpms
+
+deny contains result if {
+  some task in input.attestations[_].statement.predicate.buildConfig.tasks
+  branch := task.invocation.environment.annotations["pipelinesascode.tekton.dev/source-branch"]
+
+  not matches_any(branch)
+
+  result := lib.result_helper(rego.metadata.chain(), ["invalid branch", branch])
+}
+
+matches_any(branch) {
+  some pattern in data.rule_data.git_branch.allowed_branch_patterns[_]
```

Comment:
We have some shared code for rule data that handles default values, config overrides, etc:
```suggestion
  some pattern in lib.rule_data("allowed_branch_patterns")
```